src/callback_registry.cpp

```diff
@@ -253,7 +253,28 @@ void testCallbackOrdering() {
   }
 }
 
-CallbackRegistry::CallbackRegistry() : mutex(tct_mtx_recursive), condvar(mutex) {
+CallbackRegistry::CallbackRegistry(int id)
+  : id(id), mutex(tct_mtx_recursive), condvar(mutex)
+{
+  ASSERT_MAIN_THREAD()
+}
+
+CallbackRegistry::~CallbackRegistry() {
+  ASSERT_MAIN_THREAD()
+}
+
+int CallbackRegistry::getId() const {
+  return id;
+}
+
+
+void CallbackRegistry::signal(bool recursive) {
+  Guard guard(mutex);
+  condvar.signal();
+
+  if (recursive && parent != nullptr) {
+    parent->signal();
```

Comment:
`signal(TRUE)`

Reply:
No longer a need for signaling the parent, since we have a shared lock.